rpcs3/rpcs3qt/log_frame.cpp

```diff
@@ -430,25 +430,28 @@ void log_frame::UpdateUI()
 {
 	const auto start = steady_clock::now();
 
+	const auto tty = g_fxo->get<tty_t>();
+
 	// Check TTY logs
-	while (const u64 size = std::max<s64>(0, g_tty_size.load() - (m_tty_file ? m_tty_file.pos() : 0)))
+	if (const usz new_size = (tty && tty->fd ? tty->fd.size() : 0); new_size > m_tty_size)
 	{
-		std::string buf;
-		buf.resize(size);
-		buf.resize(m_tty_file.read(&buf.front(), buf.size()));
+		// Updated TTY size value afterwards
+		usz store_tty_size = new_size;
```

Comment:
redundant copy. just use new_size for the same purpose

Reply:
Fixed.